src/Microsoft.AspNetCore.Razor.Language/DefaultRazorCSharpLoweringPhase.cs

```diff
@@ -60,6 +60,7 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
 
             var diagnostics = new List<RazorDiagnostic>();
             diagnostics.AddRange(syntaxTree.Diagnostics);
+            diagnostics.AddRange(irDocument.GetAllDiagnostics());
```

Comment:
In theory `irDocument.GetAllDiagnostics` should be a super-set. We want to forward the syntax tree's diagnostics into the IR document